### PR TITLE
Added Products API and seed data

### PR DESCRIPTION
**Pull request by laurenwest12**
**ROUTES**
- Added json parsing for middleware in server/index.js
- Added all API routes for Product model (get (single product and all products), put, post, and delete)
- Tested with Postman to ensure that routes are working correctly

**SEED**
- Created seed data for Product model (enjoy)
- Implemented Steven's function to seed data 
- Updated Juan's User seed to also utilize this function

### CODE REVIEW

**Review by samcomma — DISMISSED** [review]

File: server/api/Product.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+})
+
+router.delete('/:id', (req, res, next) => {
+    Product.destroy({
+            where: {
+                id: req.params.id
+            }
+    })
+    .then(() => res.status(204))
```
Comment (slevine111) [inline server/api/Product.js]: Thinking that might need to be res.sendStatus(204) for response to not be forever spinning? Or it went through fine in Postman?

**Review by j-e-torres — CHANGES_REQUESTED** [review]
in the delete route, it should be res.sendStatus(204), I've had issues before with it just being res.status

File: server/api/Product.js (hunk 1 above)
Comment (laurenwest12) [reply server/api/Product.js]: It went through Postman fine, but I'm fine with changing it.

**Review by slevine111 — APPROVED** [review]

**Review by j-e-torres — APPROVED** [review]
Good stuff

**Pull request merged**